app/javascript/portal/portalHelpers.js

```diff
@@ -68,6 +68,30 @@ export const updateThemeStyles = theme => {
   setPortalClass(theme);
 };
 
+export const openExternalLinksInNewTab = () => {
```

Comment:
Is there a way to do this on the backend? I think it should be handled there. I can imagine this being a performance issue for large pages with a lot of links, since this needs to run on initial page load. 

Even if we have to do this in JS, let's check this for a large page with a lot of links and see if it performs well, you can simulate a throttled CPU in Chrome, if it's slow, an alternative approach can be the following

```js
document.addEventListener('click', function(event) {
  if (event.target.tagName === 'A') {
    const link = event.target;
    const isInternalLink = link.hostname === window.location.hostname || link.href.includes(window.portalConfig.customDomain) || link.href.includes(window.portalConfig.hostURL);
    
    if (!isInternalLink) {
      link.target = '_blank';
      link.rel = 'noopener noreferrer'; // Security and performance benefits
      // Prevent default if you want to stop the link from opening in the current tab
      event.preventDefault(); 
    }
  }
});
```
This only checks links when they are clicked, rather than iterating over all links on the page. Also, reduces the amount of work done during page load, which can improve perceived performance for large pages

Reply:
This is better :) Thanks for the suggestion.